nsxt/resource_nsxt_policy_gateway_flood_protection_profile_binding.go

```diff
@@ -0,0 +1,288 @@
+/* Copyright © 2024 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	tier0s "github.com/vmware/terraform-provider-nsxt/api/infra/tier_0s"
+	t0localeservices "github.com/vmware/terraform-provider-nsxt/api/infra/tier_0s/locale_services"
+	tier1s "github.com/vmware/terraform-provider-nsxt/api/infra/tier_1s"
+	t1localeservices "github.com/vmware/terraform-provider-nsxt/api/infra/tier_1s/locale_services"
+	utl "github.com/vmware/terraform-provider-nsxt/api/utl"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/protocol/client"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+)
+
+func resourceNsxtPolicyGatewayFloodProtectionProfileBinding() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtPolicyGatewayFloodProtectionProfileBindingCreate,
+		Read:   resourceNsxtPolicyGatewayFloodProtectionProfileBindingRead,
+		Update: resourceNsxtPolicyGatewayFloodProtectionProfileBindingUpdate,
+		Delete: resourceNsxtPolicyGatewayFloodProtectionProfileBindingDelete,
+		Importer: &schema.ResourceImporter{
+			State: nsxtGatewayFloodProtectionProfileBindingImporter,
+		},
+		Schema: map[string]*schema.Schema{
+			"nsx_id":       getNsxIDSchema(),
+			"path":         getPathSchema(),
+			"display_name": getDisplayNameSchema(),
+			"description":  getDescriptionSchema(),
+			"revision":     getRevisionSchema(),
+			"tag":          getTagsSchema(),
+			"context":      getContextSchema(),
+			"profile_path": {
+				Type:         schema.TypeString,
+				Description:  "The path of the flood protection profile",
+				Required:     true,
+				ValidateFunc: validatePolicyPath(),
+			},
+			"tier0_id": {
+				Type:          schema.TypeString,
+				Description:   "The ID of the tier0 gateway",
+				Optional:      true,
+				ConflictsWith: []string{"context"},
+				ExactlyOneOf:  []string{"tier1_id"},
+			},
+			"tier1_id": {
+				Type:        schema.TypeString,
+				Description: "The ID of the tier1 gateway",
+				Optional:    true,
+			},
+			"locale_service_id": {
```

Comment:
May I suggest changing this to `parent_path`? This could be either T0 path, T1 path or locale service path.

Reply:
Sounds good. Done.